src/core/balance-sheet.ts

```diff
@@ -1,36 +1,50 @@
 import { getBalance as getBitcoindBalance } from "@services/bitcoind"
 import { lndsBalances } from "@services/lnd/utils"
 import { baseLogger } from "@services/logger"
-import { InvoiceUser, User } from "@services/mongoose/schema"
 import { ledger } from "@services/mongodb"
 
 import { WalletFactory } from "./wallet-factory"
 import { runInParallel } from "./utils"
+import { WalletInvoicesRepository } from "@services/mongoose"
+import { RepositoryError } from "@domain/errors"
+import { User } from "@services/mongoose/schema"
 
 const logger = baseLogger.child({ module: "balanceSheet" })
 
-export const updatePendingLightningTransactions = async () => {
-  // select distinct user ids from pending invoices
-  const usersWithPendingInvoices = InvoiceUser.aggregate([
-    { $match: { paid: false } },
-    { $group: { _id: "$uid" } },
-  ])
-    .cursor({ batchSize: 100 })
-    .exec()
+const isRepoError = (obj): boolean => obj instanceof RepositoryError
+
+const updatePendingLightningInvoices = async () => {
+  const walletInvoicesRepo = WalletInvoicesRepository()
+
+  const walletsWithPendingInvoices = walletInvoicesRepo.findWalletsWithPendingInvoices()
+
+  if (isRepoError(walletsWithPendingInvoices)) {
+    logger.error(
+      { error: walletsWithPendingInvoices },
+      "finish updating pending invoices with error",
+    )
+    return
+  }
 
   await runInParallel({
-    iterator: usersWithPendingInvoices,
+    iterator: walletsWithPendingInvoices,
     logger,
-    processor: async ({ _id }, index) => {
-      logger.trace("updating pending invoices for user %s in worker %d", _id, index)
-      const user = await User.findOne({ _id })
+    processor: async (walletId, index) => {
+      logger.trace(
+        "updating pending invoices for wallet %s in worker %d",
+        walletId,
+        index,
+      )
+      const user = await User.findOne({ _id: walletId })
```

Comment:
Why not use `UsersRepository` here?

Reply:
because this will not be user but wallet (after db migration)

Reply:
Ah right then `WalletRepository`. Its there too....

Reply:
yep :) just re-doing the invoiceuser centralization to be able to add missing tests and then do the migration (more safe in this way), after that I will change related implementations to the new structure